source/material_model/rheology/composite_viscous_creep.cc

```diff
@@ -0,0 +1,238 @@
+        return (creep_stress + lim_stress)/(2.*edot_ii);
+      }
+
+
+
+      template <int dim>
+      std::pair<double, double>
+      CompositeViscousCreep<dim>::compute_strain_rate_and_derivative (const double creep_stress,
+                                                                      const double pressure,
+                                                                      const double temperature,
+                                                                      const DiffusionCreepParameters diffusion_creep_parameters,
+                                                                      const DislocationCreepParameters dislocation_creep_parameters,
+                                                                      const PeierlsCreepParameters peierls_creep_parameters,
+                                                                      const double min_viscosity,
+                                                                      const double max_viscosity) const
+      {
+        const std::pair<double, double> diff_edot_and_deriv = diffusion_creep.compute_strain_rate_and_derivative(creep_stress, pressure, temperature, diffusion_creep_parameters);
+        const std::pair<double, double> disl_edot_and_deriv = dislocation_creep.compute_strain_rate_and_derivative(creep_stress, pressure, temperature, dislocation_creep_parameters);
+        const std::pair<double, double> prls_edot_and_deriv = peierls_creep.compute_strain_rate_and_derivative(creep_stress, pressure, temperature, peierls_creep_parameters);
+
+        const double strain_rate = creep_stress/(2.*max_viscosity) + (max_viscosity/(max_viscosity - min_viscosity))*(diff_edot_and_deriv.first + disl_edot_and_deriv.first + prls_edot_and_deriv.first);
+        const double strain_rate_deriv = 1./(2.*max_viscosity) + (max_viscosity/(max_viscosity - min_viscosity))*(diff_edot_and_deriv.second + disl_edot_and_deriv.second + prls_edot_and_deriv.second);
+
+        return std::make_pair(strain_rate, strain_rate_deriv);
+      }
+
+
+
+      template <int dim>
+      void
+      CompositeViscousCreep<dim>::declare_parameters (ParameterHandler &prm)
+      {
+
+        prm.declare_entry ("Minimum strain rate", "1.4e-20", Patterns::Double(0.),
+                           "Stabilizes strain dependent viscosity. Units: \\si{\\per\\second}.");
+
+        // Viscosity iteration parameters
+        prm.declare_entry ("Strain rate residual tolerance", "1e-22", Patterns::Double(0.),
+                           "Tolerance for correct diffusion/dislocation strain rate ratio.");
+        prm.declare_entry ("Maximum strain rate ratio iterations", "40", Patterns::Integer(0),
+                           "Maximum number of iterations to find the correct "
+                           "diffusion/dislocation strain rate ratio.");
+
+        // Strain rate and stress limiting parameters
+        prm.declare_entry ("Minimum viscosities", "1.e17",
```

Comment:
Same here, a number of material models declare min and max viscosities. It would be nice to have all these min/max values defined only once.

Reply:
I wonder if having the same (identical) names will actually cause issues during parsing. It seems like it should? If so, the name may need to be altered to something like `Minimum creep viscosity`

Reply:
See above; it isn't a problem. I have added a comment describing how the deal.II parameter handler works, and what is happening here.